### ISSUE TITLE
Upgrading from v0.1.0-beta17 to v0.1.0-beta19 with Typescript - parent is "read-only" Error

### ISSUE THREAD

**Issue by mkharibalaji** [prevwong/craft.js#255]
**Describe the bug**
While Adding a Component with Multiple Elements (Ex: Card Component in Basic Example via Typescript) updating from v0.1.0-beta17 to v0.1.0-beta19 with typescript is throwing t he below Error which was working fine in v0.1.0-beta17.

Note :
       The error is not throwing when it is in JavaScript.

"TypeError: "parent" is read-only
addLinkedNodeFromTree
node_modules/@craftjs/core/dist/esm/index.js:15"

**To Reproduce**
     Basic Example Container and Card (In my case AttribGroup) shown as below in the figure & recording
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/5992474/121449486-899e8e80-c9b7-11eb-8c40-cf91f82108e9.png">

**Expected behavior**
On Dragging & Dropping it should work in Typescript as well.

**Screenshots**

https://user-images.githubusercontent.com/5992474/121450109-b69f7100-c9b8-11eb-9c66-4544a46cd0b0.mov

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| craft.js         | v0.1.0-beta19
| React            | 16.13.1
| TypeScript       | 4.0.2
| Browser          | Firefox
| npm/Yarn         | Yarn (1.22.10)
| Operating System | MacOsX

**Comment by prevwong** [comment]
Hi @mkharibalaji, I can't quite reproduce this issue in the basic example. Could you open a CodeSandbox demonstrating the bug, that would be really helpful!

**Comment by x0k** [comment]
I have the same issue https://codesandbox.io/s/craftjs-error-g4cwv

**Comment by achendrick** [comment]
Just had the same error as @x0k.
My setup is a pretty complicated but if needed I can look into creating a CodeSandbox.

I fixed it with `npm i --save-exact @craftjs/utils@0.1.0-beta.17`

**Comment by x0k** [comment]
I found, what in my project error produces only in dev server from `react-scripts`, production build and dev server from `@cypress/react` with same webpack config (i use craco adapter) works normally even with `craftjs@0.1.0-beta.19` and `typescript@4.3.4`

**Comment by atishay** [comment]
Running on unminified code from the **develop** branch, I get this error here:

```
addLinkedNodeFromTree
node_modules/@craftjs/core/dist/esm/index.js:761

  758 |         deleteNode(existingLinkedNode);
  759 |     }
  760 |     parent.data.linkedNodes[id] = tree.rootNodeId;
> 761 |     tree.nodes[tree.rootNodeId].data.parent = parentId;
      | ^  762 |     state.nodes[tree.rootNodeId] = tree.nodes[tree.rootNodeId];
  763 |     addTreeToParentAtIndex(tree);
  764 | },
```

**Comment by eduludi** [comment]
I'm also facing this problem in development using `vite`.

I mentioned this a [discussion](https://github.com/prevwong/craft.js/discussions/265) and also created a CodeSandbox replicating the issue: https://codesandbox.io/s/craftjs-issue-with-nested-element-dpnsd

**Comment by arocam** [comment]
Hi, im trying to code the example of the tutorial but, i show me the same error:
TypeError: Cannot assign to read only property 'parent' of object '#<Object>'

I've tried with 0.1.0-beta.17 but same result.

Thx :slight_smile:

**Comment by shakdoesgithub** [comment]
Same issue as the rest of the folks here. 
One thing I noticed was while using 0.1.0-beta.17 it works in dev, but not in prod build.

**Comment by shakdoesgithub** [comment]
I found a solution for my issue. 
I noticed that when I used **setProp** it was causing the issue.
To advert it, I wrapped setProp with setTimeout and it solved the issue.

This could help you all:

```
setTimeout(() => {
  setProp((prop) => (prop.content = content))
}, 50)
```

**Comment by throne1986** [comment]
Guys, me too I am having the same problem with javascript not typescript any solution out there?

**Comment by throne1986** [comment]
> Hi, im trying to code the example of the tutorial but, i show me the same error:
> TypeError: Cannot assign to read only property 'parent' of object '#'
> 
> I've tried with 0.1.0-beta.17 but same result.
> 
> Thx 🙂

did you solve the problem?

**Comment by throne1986** [comment]
> Hi @mkharibalaji, I can't quite reproduce this issue in the basic example. Could you open a CodeSandbox demonstrating the bug, that would be really helpful!

Hi did you solve this problem?

**Comment by prevwong** [comment]
This should now be fixed in 0.1.0-beta.20 💯

**Comment by mkharibalaji** [comment]
Thanks @prevwong
